code/game/turfs/open/floor.dm

```diff
@@ -241,27 +243,46 @@
 			return TRUE
 		if(RCD_LADDER)
 			to_chat(user, "<span class='notice'>You build a ladder.</span>")
-			var/obj/structure/ladder/L = new(src)
-			L.anchored = TRUE
+			var/obj/structure/ladder/Ladder = new(src)
+			Ladder.anchored = TRUE
 			return TRUE
 		if(RCD_AIRLOCK)
-			if(locate(/obj/machinery/door/airlock) in src)
+			if(locate(/obj/machinery/door) in src)
 				return FALSE
+			if(ispath(the_rcd.airlock_type, /obj/machinery/door/window))
+				to_chat(user, "<span class='notice'>You build a windoor.</span>")
+				var/obj/machinery/door/window/new_window = new the_rcd.airlock_type(src, user.dir, the_rcd.airlock_electronics?.unres_sides)
+				if(the_rcd.airlock_electronics)
+					new_window.name = the_rcd.airlock_electronics.passed_name || initial(new_window.name)
```

Comment:
(The reason you sanitise it on the way in, is because otherwise you get situations like this where the sanitisation is completely missed)